WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -252,13 +252,17 @@ def admin_edit
     @competition = Competition.find(params[:id])
     @competition_admin_view = true
     @competition_organizer_view = false
+    @nearby_competitions = @competition.nearby_competitions(Competition::NEARBY_DAYS_WARNING, Competition::NEARBY_DISTANCE_KM_WARNING)[0, 10]
+    @nearby_competitions = @nearby_competitions.select(&:isConfirmed?) unless current_user.can_view_hidden_competitions?
     render :edit
   end
 
   def edit
     @competition = Competition.find(params[:id])
     @competition_admin_view = false
     @competition_organizer_view = true
+    @nearby_competitions = @competition.nearby_competitions(Competition::NEARBY_DAYS_WARNING, Competition::NEARBY_DISTANCE_KM_WARNING)[0, 10]
+    @nearby_competitions = @nearby_competitions.select(&:isConfirmed?) unless current_user.can_view_hidden_competitions?
```

Comment:
Since there are 2 copies of this code now, maybe create a `nearby_competitions` method that both `edit` and `admin_edit` can call?